File: Tools/split.pl
#! /usr/bin/perl
use 5.010;
use utf8;
use warnings;
use strict;

my $srt_file = shift(@ARGV);
open SRT, '<:encoding(UTF-8)', $srt_file;

my @text = <SRT>;

close SRT;

if (@ARGV) {
    $srt_file = shift(@ARGV);
}

open NEWSRT, '>:encoding(UTF-8)', $srt_file;

foreach my $line (@text) {
    chomp($line);
    $line =~ s/((?:[,.?!;:])|(?:-+))\s/$1\n\n\n\n/g;
    print NEWSRT $line;
}

close NEWSRT;

File: Tools/split.rb
#!/usr/bin/env ruby
# encoding: utf-8
require_relative 'util'

def help
  doc = <<-HELP

  HELP
end


filename = ARGV.first

backup_file(filename)

puts File.open(filename).each_line.map do |line|
  line.gsub! "\n", ""
  line.gsub! ". ", ".\n"
  line.strip!
  line
end.join